src/async_wrap.h

```diff
@@ -155,10 +160,12 @@ class AsyncWrap : public BaseObject {
   inline double get_trigger_async_id() const;
 
   void AsyncReset(v8::Local<v8::Object> resource,
-                  double execution_async_id = -1,
+                  bool skip_destroy,
```

Comment:
nit: I think this would be a good candidate for a two-valued enum, so that the argument’s meaning is clearer at the call sites (similar to e.g. `CatchMode` in `node_errors.h`)

Reply:
I revisited this change and reverted it. Instead I created a new constructor to be used for reuseable resources as I think the decision to reuse is anyway at creation time. Besides that it matches better to the requirement to have some sort of init for such resources.
This change allowed me to move the promise specific constructor back into private.